apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -212,17 +217,21 @@ defmodule Aecore.Chain.Worker do
       end)
   end
 
-  defp get_blocks(blocks_acc, next_block_hash, size) do
+  defp get_blocks(blocks_acc, next_block_hash, final_block_hash, size) do
     cond do
       size > 0 ->
         case(GenServer.call(__MODULE__, {:get_block, next_block_hash})) do
           {:error, _} -> blocks_acc
           block ->
-            updated_block_acc = [block | blocks_acc]
+            updated_blocks_acc = [block | blocks_acc]
             prev_block_hash = block.header.prev_hash
             next_size = size - 1
 
-            get_blocks(updated_block_acc, prev_block_hash, next_size)
+            if(next_block_hash != final_block_hash) do
```

Comment:
Style: remove parenthesis inside `if ... do`
Suggestion: you can use `mix credo` to find such mistakes.